### PR TITLE
get_sad() Parameter rearrangement

### PR DESCRIPTION
**Pull request by aditj**
The [comment](https://github.com/xiph/rav1e/pull/1111#issuecomment-472889389) points out that the order of parameters in the get_sad() function is not as per the usual norm of WxH
This pull tries to solve that.

### CODE REVIEW

File: src/me.rs [hunk 1]
```diff
@@ -1029,27 +1028,27 @@ pub mod test {
...
       (BLOCK_128X128, 1689792),
-      (BLOCK_4X16, 8680),
       (BLOCK_16X4, 6664),
-      (BLOCK_8X32, 31056),
+      (BLOCK_4X16, 8680),
       (BLOCK_32X8, 27600),
-      (BLOCK_16X64, 93344),
+      (BLOCK_8X32, 31056),
       (BLOCK_64X16, 116384),
+      (BLOCK_16X64, 93344),
```
Comment (rom1v) [inline src/me.rs]: Switching the parameters in the method should not require to change this code.

File: src/me.rs [hunk 2]
```diff
@@ -167,8 +167,7 @@ mod nasm {
 
   #[inline(always)]
   pub fn get_sad<T: Pixel>(
-    plane_org: &PlaneSlice<'_, T>, plane_ref: &PlaneSlice<'_, T>, blk_h: usize,
-    blk_w: usize, bit_depth: usize
+    plane_org: &PlaneSlice<'_, T>, plane_ref: &PlaneSlice<'_, T>, blk_w: usize, blk_h: usize,bit_depth: usize
```
Comment (rom1v) [inline src/me.rs]: You also need to switch the parameters for all callers.

File: src/me.rs (hunk 1 above)
Comment (aditj) [reply src/me.rs]: I changed it because the previous pull swapped them  , but yeah I'll revert the changes.

File: src/me.rs (hunk 2 above)
Comment (aditj) [reply src/me.rs]: cool :+1:

File: src/me.rs [hunk 3]
```diff
@@ -1028,28 +1027,28 @@ pub mod test {
...
+        (BLOCK_64X64, 438912),
+        (BLOCK_128X64, 1016768),
+        (BLOCK_64X128, 654272),
+        (BLOCK_128X128, 1689792),
+        (BLOCK_16X4, 6664),
+        (BLOCK_4X16, 8680),
+        (BLOCK_32X8, 27600),
+        (BLOCK_8X32, 31056),
+        (BLOCK_64X16, 116384),
+        (BLOCK_16X64, 93344),
```
Comment (rom1v) [inline src/me.rs]: You can kept this exactly like it was. This is an unnecessary diff (and the blocks are not in the same order as in the definition of `BlockSize`).

File: src/me.rs (hunk 2 above)
Comment (rom1v) [inline src/me.rs]: missing space aftter `,`

Comment (aditj) [reply src/me.rs]: Theres a space in my [branch](https://github.com/aditj/rav1e/blob/8cd090aefc75a53272972a1dd0adaa5c6fb5ea3f/src/me.rs#L170) from which this pull has been requested :confused:

Comment (rom1v) [reply src/me.rs]: Hmm, no, there isn't :wink:

https://github.com/aditj/rav1e/blob/8cd090aefc75a53272972a1dd0adaa5c6fb5ea3f/src/me.rs#L170

Comment (aditj) [reply src/me.rs]: Git is dark 🤯 .
Anyways I'll add the space , sorry :) .

**Comment by rom1v** [comment]
> This branch cannot be rebased safely

Please rebase (put the whole changes in one commit on `master`).

**Comment by aditj** [comment]
@rom1v its showing this [error](https://travis-ci.org/xiph/rav1e/jobs/519093646#L3261) , I dont know how it is related to my changes :/

**Comment by rom1v** [comment]
> I dont know how it is related to my changes :/

These are new warnings added to clippy with the latest Rust version. Fixed by https://github.com/xiph/rav1e/pull/1206.

**Comment by aditj** [comment]
> It's better once rebased on master  

I did rebase the changes on master is it okay now?

File: src/me.rs [hunk 4]
```diff
@@ -209,34 +209,33 @@ mod nasm {
...
+    plane_org: &PlaneSlice<'_, T>, plane_ref: &PlaneSlice<'_, T>, blk_w: usize, blk_h: usize, bit_depth: usize
   ) -> u32 {
     #[cfg(all(target_arch = "x86_64", feature = "nasm"))]
     {
       if mem::size_of::<T>() == 2 && is_x86_feature_detected!("ssse3") && blk_h >= 4 && blk_w >= 4 {
         return unsafe {
           let plane_org = &*(plane_org as *const _ as *const PlaneSlice<'_, u16>);
           let plane_ref = &*(plane_ref as *const _ as *const PlaneSlice<'_, u16>);
-          sad_hbd_ssse3(plane_org, plane_ref, blk_h, blk_w, bit_depth)
+          sad_hbd_ssse3(plane_org, plane_ref, blk_w, blk_h, bit_depth)
```
Comment (barrbrain) [inline src/me.rs]: The parameters of `sad_hbd_ssse3` are still in `(h, w)` order.

File: src/me.rs [hunk 5]
```diff
@@ -209,34 +209,33 @@ mod nasm {
...
         };
       }
       if mem::size_of::<T>() == 1 && is_x86_feature_detected!("avx2") && blk_h >= 4 && blk_w >= 4 {
         return unsafe {
           let plane_org = &*(plane_org as *const _ as *const PlaneSlice<'_, u8>);
           let plane_ref = &*(plane_ref as *const _ as *const PlaneSlice<'_, u8>);
-          sad_avx2(plane_org, plane_ref, blk_h, blk_w)
+          sad_avx2(plane_org, plane_ref, blk_w, blk_h)
```
Comment (barrbrain) [inline src/me.rs]: The parameters of `sad_avx2` are still in `(h, w)` order.

File: src/me.rs [hunk 6]
```diff
@@ -209,34 +209,33 @@ mod nasm {
...
         };
       }
       if mem::size_of::<T>() == 1 && is_x86_feature_detected!("sse2") && blk_h >= 4 && blk_w >= 4 {
         return unsafe {
           let plane_org = &*(plane_org as *const _ as *const PlaneSlice<'_, u8>);
           let plane_ref = &*(plane_ref as *const _ as *const PlaneSlice<'_, u8>);
-          sad_sse2(plane_org, plane_ref, blk_h, blk_w)
+          sad_sse2(plane_org, plane_ref, blk_w, blk_h)
```
Comment (barrbrain) [inline src/me.rs]: The parameters of `sad_sse2` are still in `(h, w)` order.

File: src/me.rs (hunk 4 above)
Comment (aditj) [reply src/me.rs]: Should I reorder the parameters or the arguments?

Comment (rom1v) [reply src/me.rs]: I think everything should be in (width, height) order.
